modules/configuration/src/main/java/org/apache/ignite/internal/configuration/util/ConfigurationSerializationUtil.java

```diff
@@ -0,0 +1,409 @@
+                    buf.putShort(s);
+                }
+
+                return buf.array();
+            }
+
+            case INT | ARRAY: {
+                int[] ints = (int[]) value;
+
+                ByteBuffer buf = allocateBuffer(1 + 4 * ints.length);
+
+                buf.put((byte) (INT | ARRAY));
+
+                for (int n : ints) {
+                    buf.putInt(n);
+                }
+
+                return buf.array();
+            }
+
+            case LONG | ARRAY: {
+                long[] longs = (long[]) value;
+
+                ByteBuffer buf = allocateBuffer(1 + 8 * longs.length);
+
+                buf.put((byte) (LONG | ARRAY));
+
+                for (long n : longs) {
+                    buf.putLong(n);
+                }
+
+                return buf.array();
+            }
+
+            case CHAR | ARRAY: {
+                char[] chars = (char[]) value;
+
+                ByteBuffer buf = allocateBuffer(1 + 2 * chars.length);
+
+                buf.put((byte) (CHAR | ARRAY));
+
+                for (char c : chars) {
+                    buf.putChar(c);
+                }
+
+                return buf.array();
+            }
+
+            case FLOAT | ARRAY: {
+                float[] floats = (float[]) value;
+
+                ByteBuffer buf = allocateBuffer(1 + 4 * floats.length);
+
+                buf.put((byte) (FLOAT | ARRAY));
+
+                for (float f : floats) {
+                    buf.putFloat(f);
+                }
+
+                return buf.array();
+            }
+
+            case DOUBLE | ARRAY: {
+                double[] doubles = (double[]) value;
+
+                ByteBuffer buf = allocateBuffer(1 + 8 * doubles.length);
+
+                buf.put((byte) (DOUBLE | ARRAY));
+
+                for (double d : doubles) {
+                    buf.putDouble(d);
+                }
+
+                return buf.array();
+            }
+
+            case STRING | ARRAY: {
+                String[] strings = (String[]) value;
+
+                byte[][] strBytes = Arrays.stream(strings).map(s -> s.getBytes(StandardCharsets.UTF_8)).toArray(byte[][]::new);
+
+                int totalSize = Arrays.stream(strBytes).mapToInt(bytes -> bytes.length).sum();
+
+                ByteBuffer buf = allocateBuffer(1 + 4 * strBytes.length + totalSize);
+
+                buf.put((byte) (STRING | ARRAY));
+
+                for (int i = 0; i < strings.length; i++) {
+                    buf.putInt(strings[i].length());
+
+                    buf.put(strBytes[i]);
+                }
+
+                return buf.array();
+            }
+
+            default:
+                throw new IllegalStateException(value.getClass().getName());
+        }
+    }
+
+    /**
+     * Converts byte array into a configuration value.
+     *
+     * @param bytes Serialized value.
+     * @return Deserialized configuration value.
+     */
+    public static Serializable fromBytes(byte[] bytes) {
```

Comment:
Why do you still need to return `Serializable`?

Reply:
could be Object, IDK

Reply:
Serializable is stated in every usage of the method, I'll leave it